app/database/queries/breeze_buddy/analytics.py

```
@@ -144,6 +144,11 @@ def build_analytics_where_clause(
         values.append(filters["request_id"])
         conditions.append(f"lct.request_id = ${len(values) + value_offset}")
 
+    # Provider filter (list of strings)
+    if "provider" in filters and filters["provider"]:
```

Ruff (0.14.10)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)